### PR TITLE
website: Combine fetcher and poster files, update all references to use axios for both GET and POST

### PR DESCRIPTION
**Pull request by rjmacarthy**
Hi there,

Firstly, thank you for the amazing work you are doing it's truly inspirational and I'm happy to help if I can. After browsing the front-end code I propose a change. I've made some updates to the application in this PR and I wanted to share them with you. 

I've combined the `fetcher` and `poster` files into one file, `api.ts`, and standardised the use of `axios` for both GET and POST requests. This should simplify the code and make it easier to maintain going forward. I've also updated all references to these methods throughout the codebase.  I noticed that data is run through `JSON.stringify(arg)` I've removed the need for it.  By creating an axios api instance we can setup an interceptor which controls the response and reject in one place.

Please let me know if you have any questions or concerns, and I'll be happy to address them.

Thank you

### CODE REVIEW

File: website/src/components/Messages/MessageTableEntry.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { FlaggableElement } from "src/components/FlaggableElement";
 interface Message {
   text: string;
   id: string;
-  message_id: string;
+  frontend_message_id: string;
```
Comment (fozziethebeat) [inline website/src/components/Messages/MessageTableEntry.tsx:9]: You'll need to merge, I just merged in a change that I think fixes this.

**Review by fozziethebeat — APPROVED** [review]
Thank you! I like this cleanup.

You'll need to merge against main as I think you fixed a bug someone else just fixed.

**Comment by fozziethebeat** [comment]
Also, from the build failure it looks like there's still an import site for `src/lib/fetcher`.  That'll need to be fixed too.

**Review by AbdBarho — APPROVED** [review]

File: website/src/components/Messages/MessageTableEntry.tsx (hunk 1 above)
Comment (rjmacarthy) [reply website/src/components/Messages/MessageTableEntry.tsx]: Thank you very much for your review.

**Comment by rjmacarthy** [comment]
> Also, from the build failure it looks like there's still an import site for `src/lib/fetcher`. That'll need to be fixed too.

Thank you for your review, I have addressed this and updated my pull request.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
The cypress contract tests break due to something changed in `oassist_api_client.ts` Can you investigate?

**Comment by fozziethebeat** [comment]
If possible, maybe revert the changes to the `oassist_api_client.ts`.  That's used only on the API routes and not on the client side so it's actually not using axios, it's using `node-fetch`.  That library is working well enough and doesn't need refactoring.  But our `useSWR` callsites _definitely_ need cleaning up.

**Comment by rjmacarthy** [comment]
> If possible, maybe revert the changes to the `oassist_api_client.ts`. That's used only on the API routes and not on the client side so it's actually not using axios, it's using `node-fetch`. That library is working well enough and doesn't need refactoring. But our `useSWR` callsites _definitely_ need cleaning up.

Thank you. I have reverted the change to this file and updated the code to remove the dependency of environment variables for non proxied api calls.

**Review by fozziethebeat — APPROVED** [review]
Thank you for the changes! everything passes!

**Pull request merged**